File: src/sources/reuters.ts
import Parser from 'rss-parser';
import { BaseSource, RawArticle } from './base';
import { logger } from '@/lib/logger';

const parser = new Parser({ timeout: 15000, maxRedirects: 5 });

// Reuters 공개 RSS 2023년 폐지 → STAT News + BBC Health로 대체
const FEEDS = [
  'https://www.statnews.com/feed/',
  'https://feeds.bbci.co.uk/news/health/rss.xml',
];

export class ReutersSource extends BaseSource {
  readonly name = 'STAT News';

  async fetch(): Promise<RawArticle[]> {
    const articles: RawArticle[] = [];
    for (const feedUrl of FEEDS) {
      try {
        const feed = await parser.parseURL(feedUrl);
        for (const item of feed.items) {
          if (!item.title || !item.link) continue;
          articles.push({
            source: this.name,
            title: this.stripHtml(item.title),
            url: item.link,
            published_at: this.parseDate(item.pubDate ?? item.isoDate),
            summary: item.contentSnippet ?? '',
            language: this.language,
          });
        }
      } catch (e) {
        logger.warn(`[STAT News] 피드 오류 ${feedUrl}: ${e}`);
      }
    }
    return articles;
  }
}

File: src/sources/reuters.py
import feedparser
from .base import BaseSource, RawArticle


class ReutersSource(BaseSource):
    name = "Reuters"
    language = "en"
    FEEDS = [
        "https://feeds.reuters.com/reuters/healthNews",
        "https://feeds.reuters.com/reuters/scienceNews",
    ]

    async def fetch(self) -> list[RawArticle]:
        articles = []
        for url in self.FEEDS:
            try:
                feed = feedparser.parse(url)
                for entry in feed.entries:
                    articles.append(RawArticle(
                        source=self.name,
                        title=entry.get("title", "").strip(),
                        url=entry.get("link", ""),
                        published_at=self._safe_date(entry.get("published")),
                        summary=entry.get("summary", ""),
                        language=self.language,
                    ))
            except Exception as e:
                from src.utils.logger import logger
                logger.warning(f"[Reuters] 피드 오류 {url}: {e}")
        return articles
